### PR TITLE
Eleventy sandbox

### LINKED ISSUE #84: Migrate sandbox away from Jekyll
Jekyll has issues with ruby dependencies which has caused setup issues on different platforms. Plugins and dependencies are also not very well maintained or up-to-date.

### PR DESCRIPTION
**Pull request by mejiaj**
# Migrate 11ty (from jekyll)

Fixed [preview link](https://federalist-d5e7c07c-6ffa-4b8e-935f-49a7df24a505.sites.pages.cloud.gov/preview/uswds/uswds-sandbox/jm-eleventy-sandbox/)

_⚠️ ~CG-Pages build is failing because the site engine is set to `Jekyll`. Once we switch to `Node.js` we should be OK.~ _

Jekyll is a great tool, but it's ruby dependencies can cause setup issues on different platforms. They're also not very well maintained or up-to-date. This PR gets rid of Jekyll and it's dependencies in favor of a JS-based static site generator (eleventy). Closes #84.

## Benefits

- **Similar templating to Jekyll.**  front matter, liquid templating and markdown are still supported (as well as nunjucks).
- **Similar file structure to Jekyll.** Still uses `_data, _includes, _layouts` for managing data and templates. The output still builds HTML to `_site` directory.
- **Improved dependencies.** All dependencies are in node and requires less boilerplate to setup. As well as more up-to-date than ruby and jekyll plugins.
- **Faster cloud page builds.** Page builds have gone down from 4-5min to 1-2. Will keep an eye on this, but so far it's pretty nice.

## Differences

- **Site structure is flattened.** Jekyll creates one-off pages from `pages/`. In 11ty you'd make a top-level page next to your index.
- **Config is handled in JS.** Configuration is now set in `.eleventy.js`. Learn more in [11ty documentation](https://www.11ty.dev/docs/config/). Currently there are settings for:
  - Copying assets to site build
  - Setting pathPrefix so assets and URLs work in CG-Pages
  - Setting template formats to support `md, njk, liquid`
  - Setting layout and data directories
- **Jekyll-specific filters no longer supported.** Setting absolute and relative url's is handled by `url` filter. Haven't found an equivalent to `normalize_whitespace`.
- **Includes use quotes.** For example, including skipnav is now `{% include "skipnav.html" %}`.
- **Defining layout uses quotes.** Use quotes when defining layouts in front matter. For example `layout: "extended-prose.html"`.

## Related
https://github.com/uswds/uswds-tutorial/issues/12

## How to test
- [x] Install succesfully with `npm install`
- [x] Run local setup with `npm start`
- [x] Build without errors to `_site` with `npm run build`
- [x] Original templates should be included

### CODE REVIEW

**Comment by mejiaj** [comment]
How was the install process compared to Jekyll? Those screenshots looks mostly correct. I'll look into the missing images. Thanks @amyleadem.

**Review by mahoneycm — COMMENTED** [review]
Built very quickly for me!

image

Images loaded appropriately for me
image

**Comment by amyleadem** [comment]
@mejiaj I removed the _site directory and did a fresh install, and now my assets are happy. 
image

File: .eleventy.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+module.exports = function (eleventyConfig) {
+  eleventyConfig.addPassthroughCopy("assets");
```
Comment (mejiaj) [inline .eleventy.js:2]: `addPassthroughCopy` allows us to move over any assets or directories we need to the final site build.

File: .eleventy.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+  return {
+    pathPrefix: pathPrefix,
+    templateFormats: ["md", "njk", "html", "liquid"],
+    markdownTemplateEngine: "liquid",
+    htmlTemplateEngine: "liquid",
+    dir: {
+      layouts: "_layouts",
+      data: "_data",
+    },
```
Comment (mejiaj) [inline .eleventy.js:17]: The layouts and data worked without these settings, but making them explicit for clarity.

File: .eleventyignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+README.md
+_drafts/
```
Comment (mejiaj) [inline .eleventyignore:2]: This file tells 11ty to **not** build these pages.

File: .nvmrc [hunk 4]
```diff
@@ -1 +1 @@
-16
\ No newline at end of file
+18
```
Comment (mejiaj) [inline .nvmrc:1]: 18 is the latest LTS version. Didn't experience any conflicts with dependencies.

File: _data/site.json [hunk 5]
```diff
@@ -0,0 +1,6 @@
+{
+  "title": "USWDS Sandbox",
+  "baseurl": "",
```
Comment (mejiaj) [inline _data/site.json:3]: `baseurl` is from previous data. Need to consider removing to avoid confusion in eleventy.

**Comment by mejiaj** [comment]
Snyk is failing because it's trying to check dependencies in the now deleted Gemfile.lock.

I've run snyk locally and got:

```bash
✔ Tested 9 dependencies for known issues, no vulnerable paths found.
```

**Review by thisisdano — APPROVED** [review]

**Pull request merged**